src/taipy/core/config/migration_config.py

```diff
@@ -0,0 +1,188 @@
+            data_node_config (Union[DataNodeConfig, DataNodeId]): The data node configuration or the `id` of
+                the data node config that need to migrate.
+            migration_fct (Callable): Migration function that takes a DataNode entity from `target_version` as input
+                and returns a compatible DataNode with the next production version.
+            **properties (Dict[str, Any]): A keyworded variable length list of additional arguments.
+        Returns:
+            `MigrationConfig^`: The Migration configuration.
+        """
+        return MigrationConfig.__add_migration_function(target_version, data_node_config, migration_fct, **properties)
+
+    @staticmethod
+    def _add_task_migration_function(
+        target_version: str,
+        task_config: Union[TaskConfig, TaskId],
+        migration_fct: Callable,
+        **properties,
+    ):
+        """Add a task migration function to the configuration.
+
+        Parameters:
+            target_version (str): The version that entities are migrated to.
+            task_config (Union[TaskConfig, TaskId]): The task configuration or the `id` of
+                the task config that need to migrate.
+            migration_fct (Callable): Migration function that takes a DataNode entity from `target_version` as input
+                and returns a compatible DataNode with the next production version.
+            **properties (Dict[str, Any]): A keyworded variable length list of additional arguments.
+        Returns:
+            `MigrationConfig^`: The Migration configuration.
+        """
+        return MigrationConfig.__add_migration_function(target_version, task_config, migration_fct, **properties)
+
+    @staticmethod
+    def _add_pipeline_migration_function(
+        target_version: str,
+        pipeline_config: Union[PipelineConfig, PipelineId],
+        migration_fct: Callable,
+        **properties,
+    ):
+        """Add a pipeline migration function to the configuration.
+
+        Parameters:
+            target_version (str): The version that entities are migrated to.
+            pipeline_config (Union[PipelineConfig, PipelineId]): The pipeline configuration or the `id` of
+                the pipeline config that need to migrate.
+            migration_fct (Callable): Migration function that takes a DataNode entity from `target_version` as input
+                and returns a compatible DataNode with the next production version.
+            **properties (Dict[str, Any]): A keyworded variable length list of additional arguments.
+        Returns:
+            `MigrationConfig^`: The Migration configuration.
+        """
+        return MigrationConfig.__add_migration_function(target_version, pipeline_config, migration_fct, **properties)
+
+    @staticmethod
+    def _add_scenario_migration_function(
+        target_version: str,
+        scenario_config: Union[ScenarioConfig, ScenarioId],
```

Comment:
```suggestion
        scenario_config: Union[ScenarioConfig, str],
```

ScenarioId is an alias for the id of a scenario. not a section id (ScenarioConfigId).